addons/switch_focus/box_list.lua

```diff
@@ -0,0 +1,51 @@
+local box_list = {
+    _orderd = {},
+    _byValue = {},
+
+    len = function(self)
+        return #self._orderd
+    end,
+
+    add = function(self, value)
+        if self._byValue[value] then
+            return
+        end
+
+        table.insert(self._orderd, value)
+        table.sort(self._orderd)
+
+        self._byValue = {}
+        for k, v in ipairs(self._orderd) do
+            self._byValue[v] = k
+        end
```

Comment:
Why does the table need to be cleared and repopulated? Couldn't you just assign the new value? Same in `remove`, there you coul just remove the value.

Reply:
the order could change, meaning the value for each name to index could change.
for example: i have Nifim logged in then Vanim joins so `Nifim = 1, Vanim = 2` then i log into Jotim when we add him the indexes become `Jotim = 1, Nifim = 2, Vanim = 3` he causes a shift in the whole array so clearing out the table for each addition or removal is necessary.